.github/workflows/linux-ci-sonarcloud.yml

```diff
@@ -0,0 +1,49 @@
+name: Linux CI SonarCloud
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install system dependencies
```

Comment:
can you check if we can reuse these steps? it's duplicated a few times

Reply:
I can move it to the main CI file but it will increase the time of the CI - let's investigate if we can have re-usable part yes.